Update: package, php version, ci, everything !

shulard opened the pull request.

shulard committed:
chore: update dependencies

New PHP version, latest package updates…

shulard committed:
chore(ci): add PHP8.4 && 8.5

shulard committed:
chore(github): configure dependabot properly

shulard committed:
chore(phpstan): fix latest issues detection

shulard committed:
chore(php): fix implicit nullable parameters

Not compatible with PHP8.4, nullable parameters must use
nullable types.

shulard committed:
chore: update composer to 2.9 and PHP to 8.5

shulard committed:
chore(test): update PHPUnit to 13 and migrate tests

The tests weren't using the latest PHPUnit changes
(annotation, stub…).

Also all the detected warning / deprecations / errors
have been fixed.

Rector has been installed to help migrated PHPUnit code.

shulard commented:
@tgndrd can you take a look at this one ? It's almost a huge update :smile:. I invited you in the organization.

tgndrd commented on .github/workflows/continuous-integration.yml:
```diff
@@ -92,7 +94,7 @@ jobs:
           - "locked"
         php-version:
-          - "8.3"
+          - "8.4"
```
why dont you add php 8.5 here also ?

shulard replied:
Here I'm not sure it's relevant to test on multiple PHP versions since that's only to check php-cs-fixer (the fixes aren't different based on a specific PHP version).

I think it's still better to use 8.5 instead of 8.4.

tgndrd commented on .github/workflows/test-coverage.yml:
```diff
@@ -18,7 +18,7 @@ jobs:
         with:
           coverage: "pcov"
-          php-version: "8.3"
+          php-version: "8.4"
```
same as above

shulard replied:
Same here, I think it's better to use 8.5 instead of 8.4 however it's not relevant to generate the test coverage on multiple PHP versions.

tgndrd commented on src/Bridge/LeagueOpenAPIValidation/Exception/ValidationExceptionMapper.php:
```diff
@@ -50,7 +50,11 @@ private function mapSchemaMismatch(SchemaMismatch $lastError): DataSchemaExcepti
-            $crumbs = implode('.', $breadCrumb->buildChain());
+            $crumbs = array_reduce(
+                $breadCrumb->buildChain(),
+                static fn (string $carry, mixed $value): string => ($carry !== '' ? $carry . '.' : '') . (\is_int($value) || \is_string($value) ? (string) $value : ''),
```
maybe explode the double ternary to improve readability

```php
                static function (string $carry, mixed $value): string {
                    $strValue = \is_int($value) || \is_string($value) ? (string) $value : '';

                    return $carry !== '' ? $carry . '.' . $strValue : $strValue;
                },
```

shulard replied:
Yes that's better, I fixed it!

tgndrd commented:
:heavy_check_mark: 

approved

shulard committed:
chore(test): add missing test cases

Also reinforce typing for headers. We never accept "mixed"
as header so we type as array.
